lms/static/scripts/frontend_apps/types.d.ts

```diff
@@ -1,7 +1,40 @@
 export {};
 
 declare global {
+  interface NavigationEventMap {
+    navigate: Event;
+    navigatesuccess: Event;
+    navigateerror: ErrorEvent;
+  }
+
+  interface Navigation extends EventTarget {
+    addEventListener<K extends keyof NavigationEventMap>(
+      type: K,
+      listener: (this: Navigation, ev: NavigationEventMap[K]) => any,
+      options?: boolean | AddEventListenerOptions,
+    ): void;
+    addEventListener(
+      type: string,
+      listener: EventListenerOrEventListenerObject,
+      options?: boolean | AddEventListenerOptions,
+    ): void;
+    removeEventListener<K extends keyof NavigationEventMap>(
+      type: K,
+      listener: (this: Navigation, ev: NavigationEventMap[K]) => any,
+      options?: boolean | EventListenerOptions,
+    ): void;
+    removeEventListener(
+      type: string,
+      listener: EventListenerOrEventListenerObject,
+      options?: boolean | EventListenerOptions,
+    ): void;
+  }
+
   interface Window {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API
+    // https://github.com/lukewarlow/navigation-api-types
+    readonly navigation?: Navigation;
+
```

Comment:
This brings the minimum types required to add and remove navigation events.

I took it from https://github.com/lukewarlow/navigation-api-types, but did not install the whole package as this is just a dev-env affordance.

Reply:
The package contains only 184 lines with no dependencies so you might as well add it as a development dependency. That will have the benefit of documenting where the types come from and make it easier to remove in future when we no longer need them.